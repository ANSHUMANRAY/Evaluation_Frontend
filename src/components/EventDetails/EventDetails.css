
.eventDetailsImg{
    width: 45vw;
    border-radius: 50px 50px 0 0;
}

.eventDetails{
    width: 45vw;
    height: fit-content;
    border-radius: 50px;
    margin: 10px auto;
    text-align: center;
    border: 1px solid #ccc;
    color: white;
}

.eventDetailStatus{
    width: 45vw;
    height: 50px;
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.checkDetail{
    width: 40px;
    height: 40px;
    margin: 0 10px;
}

.checkDetailDiv{
    display: flex;
    justify-content: space-between;
}

.detailBookmark{
    width: 40px;
    height: 40px;
}

button{
    background-color:white;
    border: none;
    color: black;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
}

