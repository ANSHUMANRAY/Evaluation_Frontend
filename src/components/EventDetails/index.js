
import React from 'react';
import axios from 'axios';
import check from '../../assets/circle-check-solid.svg';
import cross from '../../assets/circle-xmark-solid.svg';
import bookmarked from '../../assets/bookmark-solid.svg';
import bookmark from '../../assets/bookmark-regular.svg';
import { useParams } from 'react-router-dom';
import './EventDetails.css';
import { ThemeContext } from '../../contexts/themeContext';

export default function Event() {
  const {theme} = React.useContext(ThemeContext);
  const {id} = useParams();
  const [event, setEvent] = React.useState({});
  const [isBookmarked, setIsBookmarked] = React.useState(event.isBookmarked);
  const [isRegistered, setIsRegistered] = React.useState(event.isRegistered);
  React.useEffect(() => {
    axios.get(`http://localhost:8000/api/events/${id}`)
      .then((response) => {
        setEvent(response.data);
        setIsBookmarked(response.data.isBookmarked);
        setIsRegistered(response.data.isRegistered);
      });
  }, []);
  const bookmarkIcon = isBookmarked ? bookmarked : bookmark;
  const registerText = isRegistered ? 'UNREGISTER' : 'REGISTER';
  const handleBookmark = () => {
    axios.patch(`http://localhost:8000/api/events/${event.id}`, {isBookmarked: !isBookmarked})
      .then(() => {setIsBookmarked(!isBookmarked);});
  };
  const handleRegister = () => {
    axios.patch(`http://localhost:8000/api/events/${event.id}`, {isRegistered: !isRegistered})
      .then(() => {setIsRegistered(!isRegistered);});
  };
  return event ? (
    <div style={{backgroundColor: `${theme}`}} className="eventDetails">
      <img src={event.imgUrl} alt={event.name} className="eventDetailsImg" />
      <div className="eventDetailsInfo">
        <p className='eventDetailsName'>{event.name}</p>
        <p className='detailsDescription'>{event.description}</p>
        <p className='detailsVenue'>VENUE:{event.venue}</p>
        <p >DATE:{event.datetime}</p>
      </div>
      <div className="eventDetailStatus">
        <div>
          {event.isRegistered ? (
            <div className='checkDetailDiv'>
              <img className='checkDetail' src={check}/>
              <p>Registered</p>
            </div>
          ) : (event.areSeatsAvailable ? (
            <div className='checkDetailDiv'>
              <img className='checkDetail' src={cross}/>
              <p>No Seats Available</p>
            </div>
          ) : (<div></div>))}
        </div>
        <div onClick={handleBookmark}>
          <img className='detailBookmark' src={bookmarkIcon} alt='bookmark' />
        </div>
      </div>
      <button onClick={handleRegister} className='registerButton'>{registerText}</button>
    </div>
  ) : (<div>Loading...</div>);
}