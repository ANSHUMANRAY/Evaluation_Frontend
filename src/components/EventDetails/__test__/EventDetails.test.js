import React from 'react';
import {render} from '@testing-library/react';
import EventDetails from '../index';
import { ThemeContext } from '../../../contexts/themeContext';
import { mockEvent } from '../../../mocks/mockEvents';
import axios from 'axios';

jest.mock('axios');

describe('EventDetails', () => {
  it('should render correctly', () => {
    axios.get.mockResolvedValue({data: mockEvent});
    const {container} = render(<ThemeContext.Provider value={{theme: 'black', setTheme: jest.fn()}}><EventDetails/></ThemeContext.Provider>);
    expect(container).toMatchSnapshot();
  });
});