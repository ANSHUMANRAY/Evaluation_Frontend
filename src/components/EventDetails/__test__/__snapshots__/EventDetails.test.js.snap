// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails should render correctly 1`] = `
<div>
  <div
    class="eventDetails"
    style="background-color: black;"
  >
    <img
      class="eventDetailsImg"
    />
    <div
      class="eventDetailsInfo"
    >
      <p
        class="eventDetailsName"
      />
      <p
        class="detailsDescription"
      />
      <p
        class="detailsVenue"
      >
        VENUE:
      </p>
      <p>
        DATE:
      </p>
    </div>
    <div
      class="eventStatus"
    >
      <div>
        <div />
      </div>
      <div>
        <img
          alt="bookmark"
          class="bookmark"
          src="bookmark-regular.svg"
        />
      </div>
    </div>
    <button
      class="registerButton"
    >
      REGISTER
    </button>
  </div>
</div>
`;
